idaplugin/rematch/actions/settings.py

```diff
@@ -1,16 +1,25 @@
 import idaapi
 
 from . import base
+from .. import config
 
 from ..dialogs.settings import SettingsDialog
 
 
 class SettingsAction(base.Action):
   name = "&Settings"
 
-  @staticmethod
-  def activate(ctx):
-    SettingsDialog().get()
+  def activate(self, ctx):
+    self.dlg = SettingsDialog(submit_handler=self.submit)
+    self.dlg.show()
+
+  def submit(self, autocheck, autoupdate, autologin, autologout, debug):
```

Comment:
I wanted to make sure all (and only) parameters are passed, and actually defining the function parameters seemed reasonable. This also make it easier to code/read, because you know exactly which parameters are available.